src/inline-assembly.md

```diff
@@ -37,515 +52,937 @@ assert_eq!(x, 4 * 6);
+
+```rust
+# #[cfg(target_arch = "x86_64")] {unsafe{
+let mut x: i32;
+// The following lines are equivalent
+core::arch::asm!("mov rax, 5", "mov rcx, rax", out("rax") x, out("rcx") _); 
+core::arch::asm!("mov rax, 5\nmov rcx, rax", out("rax") x, out("rcx") _);  
+# }}
+```
+
+r[asm.invocation.operands]
+Each operand, other than an explicit register operand ([asm.operands.register]) shall be mentioned by at least one format_specifier in the *joined asm-string*. Explicit registers may not be referred to be a format_specifier.
+
+```rust,compile_fail
+# #[cfg(target_arch = "x86_64")] { unsafe{
+core::arch::asm!("", in(reg) 5i64);
+# }}
+# #[cfg(not(target_arch = "x86_64"))]
+# core::compile_error!("asm tests are not yet available off of x86_64"); 
+```
+
+r[asm.invocation.positional]
+A `format_specifier` that does not specify an `operand_specifier` is called a positional specifier, and refers to the `nth` successive positional operand, where `n` is `0` for the first positional specifier in the *joined asm-string* and increases by 1 for each successive positional specifier in the *joined asm-string*.
+
+```rust
+# #[cfg(target_arch = "x86_64")] { unsafe{
+let x: i32;
+core::arch::asm!("mov rax, {}", in(reg) 5i64, out("eax") x);
+# }}
+```
+
+r[asm.invocation.explicit-positional]
+A `format_specifier` that has an `operand_specifier` which is a DEC_LITERAL is called an explicit positional specifier, and refers to the `nth` successive positional operand, where `n` is the value of the DEC_LITERAL.
+
+```rust
+# #[cfg(target_arch = "x86_64")] { unsafe{
+let x: i32;
+core::arch::asm!("mov {1}, {0}", in(reg) 5i64, out(reg) x);
+# }}
+```
+
+r[asm.invocation.named]
+A `format_specifier` that has an `operand_specifier` which is an ident is called a named specifier, and refers to the named operand with the specified name.
+
+```rust
+# #[cfg(target_arch = "x86_64")] { unsafe{
+let x: i32;
+core::arch::asm!("mov {output}, {input}", input = in(reg) 5i64, output = out(reg) x);
+# }}
+```
+
+r[asm.invocation.expansion]
+If the `raw` option is not specified, the *joined asm-string* is expanded as defined in [asm.operands.expansion], replacing each `format_specifier` with the appropriate expansion for the operand. The resulting string is called the *expanded asm-string*. If the `raw` option is specified, the *expanded asm-string* is the *joined asm-string* verbatim. 
+
+r[asm.invocation.syntax]
+The syntax of the *expanded asm-string* is a subset of the GNU AS syntax for the target. Invoking the macro with a *expanded asm-string* that does not match syntax requirements is *conditionally supported* and has *assembler dependent behaviour*. Invoking a directive that is not specified by [asm.directives] is *conditionally supported* and has *assembler dependent behaviour*.
```

Comment:
Can you make sure to use American English? (Here and elsewhere in the document.)

```suggestion
The syntax of the *expanded asm-string* is a subset of the GNU AS syntax for the target. Invoking the macro with a *expanded asm-string* that does not match syntax requirements is *conditionally supported* and has *assembler dependent behavior*. Invoking a directive that is not specified by [asm.directives] is *conditionally supported* and has *assembler dependent behavior*.
```